wrangler-api/src/main/java/io/cdap/wrangler/api/lineage/Relation.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.api.lineage;
+
+import io.cdap.cdap.api.annotation.Beta;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * The <code>Relation</code> class represents a relation between a list of sources to a list of targets.
+ * The relation can have 4 different types and can be many-to-many, many-to-one, one-to-many or one-to-one.
+ *
+ * <p>A relation can be of four different types namely:</p>
+ * <ul>
+ *   <li>
+ *     <code>DROP type</code> - This type represents relation with a non-existential destination - one-to-none
+ *   type. This type is used when the column wouldn't be present after the transformation is performed.
+ *   </li>
+ *   <li>
+ *     <code>ALL</code> - This type represents a relation with many. one-to-many or many-to-many relation. This
+ *     type is generally used when the transformation is unclear about what columns would be generated.
+ *   </li>
+ *   <li>
+ *     <code>CREATE</code> - This type represents a relation in which a transformation creates a target without
+ *     having any source.
+ *   </li>
+ *   <li>
+ *     <code>STANDARD</code> - This type represents all standard relations like many-to-many, many-to-one and
+ *     one-to-many.
+ *   </li>
+ * </ul>
+ *
+ * @see Lineage
+ * @see Many
+ * @see Mutation
+ */
+@Beta
+public final class Relation implements Serializable {
+  /**
+   * Specifies the type of relations that can exist between source and target.
+   */
+  public enum Type {
+    DROP,
+    ALL,
+    CREATE,
+    STANDARD,
```

Comment:
why do we have `ALL` and `STANDARD` types?

Reply:
STANDARD means the relation has to clear-defined, the source and target has to be specified. ALL means the directive will apply changes to all the output fields, which we don't know in the directive, so we will have special logic when we generate lineage for ALL operations